hazelcast-client/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java

```diff
@@ -193,6 +194,10 @@ public HazelcastClientInstanceImpl(ClientConfig config,
             addressProviders.add(new DiscoveryAddressProvider(discoveryService));
         }
 
+        if (Boolean.valueOf(getClientConfig().getProperty(GroupProperty.DISCOVERY_SPI_ENABLED))) {
```

Comment:
@Donnerbart `DISCOVERY_SPI_ENABLED` was not defined in `ClientProperty`. I added the definition. Can you look at the PR again?  Is this the correct way?
``` 
if (clientProperties.getBoolean(ClientProperty.DISCOVERY_SPI_ENABLED)) {
       discoveryService.start();
}
```